File: examples/example-kubernetes-client-openapi-java/src/main/java/micronaut/client/PodController.java
package micronaut.client;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.kubernetes.client.openapi.api.CoreV1Api;
import io.micronaut.kubernetes.client.openapi.model.V1Pod;
import io.micronaut.kubernetes.client.openapi.model.V1PodList;
import io.micronaut.scheduling.TaskExecutors;
import io.micronaut.scheduling.annotation.ExecuteOn;
import jakarta.inject.Inject;
import jakarta.validation.constraints.NotNull;

import java.util.Map;
import java.util.stream.Collectors;

@Controller("/pods")
@ExecuteOn(TaskExecutors.BLOCKING)
public class PodController {

    @Inject
    CoreV1Api coreV1Api;

    @Get("/{namespace}/{name}")
    public String getPod(final @NotNull String namespace, final @NotNull String name) {
        V1Pod v1Pod = coreV1Api.readNamespacedPod(name, namespace, null);
        return v1Pod.getStatus().getPhase();
    }

    @Get("/{namespace}")
    public Map<String, String> getPods(final @NotNull String namespace) {
        V1PodList v1PodList = coreV1Api.listNamespacedPod(namespace, null, null, null, null, null, null, null, null, null, null, false);
        return v1PodList.getItems().stream()
                .filter(p -> p.getStatus() != null)
                .collect(Collectors.toMap(
                        p -> p.getMetadata().getName(),
                        p -> p.getStatus().getPhase()));
    }
}

File: examples/example-kubernetes-client-openapi-python/src/main/python/micronaut/client/PodController.py
from typing import Annotated

from jakarta.inject import Inject
from jakarta.validation.constraints import NotNull
from micronaut.http.annotation import Controller, Get
from micronaut.kubernetes.client.openapi.api import CoreV1Api
from micronaut.scheduling import TaskExecutors
from micronaut.scheduling.annotation import ExecuteOn


@Controller("/pods")
@ExecuteOn(TaskExecutors.BLOCKING)
class PodController:

    core_v1_api: Annotated[CoreV1Api, Inject]

    @Get("/{namespace}/{name}")
    def get_pod(self, namespace: Annotated[str, NotNull], name: Annotated[str, NotNull]) -> str:
        v1_pod = self.core_v1_api.readNamespacedPod(name, namespace, None)
        return v1_pod.getStatus().getPhase()

    @Get("/{namespace}")
    def get_pods(self, namespace: Annotated[str, NotNull]) -> dict[str, str]:
        v1_pod_list = self.core_v1_api.listNamespacedPod(namespace, None, None, None, None, None, None, None, None, None, None, None, None)
        return {
            p.getMetadata().getName(): p.getStatus().getPhase()
            for p in v1_pod_list.getItems()
            if p.getStatus() is not None
        }
